pkg/gofr/datasource/file/ftp/file.go

```diff
@@ -0,0 +1,277 @@
+func (f *ftpFile) ReadAll() (RowReader, error) {
+	if strings.HasSuffix(f.Name(), ".json") {
+		return f.createJSONReader()
+	}
+
+	return f.createTextCSVReader()
+}
+
+// createJSONReader creates a JSON reader for JSON files.
+func (f *ftpFile) createJSONReader() (RowReader, error) {
+	size, err := f.conn.FileSize(f.path)
+	if err != nil {
+		return nil, err
+	}
+
+	buffer := make([]byte, size)
+
+	_, readerError := f.Read(buffer)
+
+	if readerError != nil && !errors.Is(readerError, io.EOF) {
+		f.logger.Errorf("ReadAll Failed Failed to read json file: %v", readerError)
+		return nil, readerError
+	}
+
+	reader := bytes.NewReader(buffer)
+
+	decoder := json.NewDecoder(reader)
+
+	// Peek the first JSON token to determine the type
+	token, err := decoder.Token()
+	if err != nil {
+		return nil, err
+	}
+
+	if d, ok := token.(json.Delim); ok && d == '[' {
+		return &jsonReader{decoder: decoder, token: token}, readerError
+	}
+
+	// JSON object
+	return f.createJSONObjectReader(reader)
+}
+
+func (_ *ftpFile) createJSONObjectReader(reader *bytes.Reader) (RowReader, error) {
+	decoder := json.NewDecoder(reader)
+
+	return &jsonReader{decoder: decoder}, nil
+}
+
+// createTextCSVReader creates a text reader for reading text files.
+func (f *ftpFile) createTextCSVReader() (RowReader, error) {
+	size, err := f.conn.FileSize(f.path)
+	if err != nil {
+		return nil, err
+	}
+
+	buffer := make([]byte, size)
+
+	_, err = f.Read(buffer)
+	if err != nil {
+		f.logger.Errorf("ReadAll Failed Failed to read text file: %v", err)
+		return nil, err
+	}
+
+	reader := bytes.NewReader(buffer)
+
+	return &textReader{
+		scanner: bufio.NewScanner(reader),
+		logger:  f.logger,
+	}, err
+}
+
+// Next checks if there is another JSON object available.
+func (j *jsonReader) Next() bool {
+	return j.decoder.More()
+}
+
+// Scan decodes the next JSON object into the provided structure.
+func (j *jsonReader) Scan(i interface{}) error {
+	return j.decoder.Decode(&i)
+}
+
+// Next checks if there is another line available in the text file.
+func (f *textReader) Next() bool {
+	return f.scanner.Scan()
+}
+
+// Scan scans the next line from the text file into the provided pointer to string.
+func (f *textReader) Scan(i interface{}) error {
+	switch target := i.(type) {
+	case *string:
+		*target = f.scanner.Text()
+		return nil
+	default:
+		return errNotPointer
+	}
+}
+
+// Close closes the FTP file connection.
+func (f *ftpFile) Close() error {
+	return f.response.Close()
+}
+
+// Name returns the name of the file.
+func (f *ftpFile) Name() string {
+	return f.name
+}
+
+// Read reads data from the FTP file into the provided byte slice and updates the file offset.
+func (f *ftpFile) Read(p []byte) (n int, err error) {
+	r, err := f.conn.RetrFrom(f.path, uint64(f.offset))
+	if err != nil {
+		f.logger.Errorf("Read Failed Failed to open file with path %q : %v", f.path, err)
```

Comment:
The "failed failed" seems strange. It could be normal or not. I'm unsure.

Maybe the issue could resolved by adding an hypen `-` or a colon `:` to separate text.